### ISSUE TITLE
Display semantic information as named edges

### ISSUE THREAD

**Issue by krabina** [ProfessionalWiki/Network#43]
As a new feature, it would be great to add more semantic capabilities to the network extension.

Currenly, "only" links from one page to others are shown. There are several ways, how this can be improved.
1. Named edges. As soon as the links from one page to another are the result of a result format (lets say table of pages that are the result of a SMW query, the pages would be shown if they are of datatype page. But the name of the properties could here be shown as names of the edges. 
2. Semantics instead of just page links: As the [Semantic MediaWiki Graph extension](https://www.mediawiki.org/wiki/Extension:Semantic_MediaWiki_Graph)  shows, one could take this further and also show other properties with the values as nodes and the name of the properties as edges. This then would be an option to basically visualize what is in the Special:Browse interface instead of the links of the page. Could probably be implemented as an option for the parser function to take semantics of the page instead of just the links. Could also be an option on the Special:Network page.

![grafik](https://user-images.githubusercontent.com/4318745/122356315-f6370180-cf52-11eb-9e2a-4b35141196d4.png)
3. A network result format. Of course, another possibility would be to implement an SMW result format that then would automatically show everything the SMW query delivers.

**Comment by JeroenDeDauw** [comment]
This is quite different from showing connections between pages. There is little similarity between that in both cases a network is used for visualization. Which suggests that while both the page connection and SMW cases could use the same network visualization library (visjs), using the same code (JS, PHP & parser function) in this extension is not possible without making a mess. In other words, I strongly expect creating a new parser function (or similar) with its own implementation is the way to go. And that could just as well be in its own extension.

**Comment by cicalese** [comment]
I agree that, if implemented, this should be a separate graph capability. Network graphs can quickly become overwhelmingly complex to read, so we should resist the urge to add different types of information to the same graph. Also, in order to get the property name for edges, a completely different implementation would be necessary, because the semantic links currently come from the links table, where no such annotation exists. That being said, I do think that visualizing semantic information in a separate graph could be useful.

**Comment by krabina** [comment]
I always like the approach of making use of semantics when SMW is present and as now we have tow unmaintained network-extensios (Viki and SMW Graph), I think this extension is the ideal place to start this. Anyway, this is just a feature request and I am happy to discuss this in more detail.

**Comment by krabina** [comment]
I am closing this, since we are working on https://www.mediawiki.org/w/index.php?title=Extension:KnowledgeGraph
